app/services/reports/facility_progress_service.rb

```diff
@@ -82,15 +83,86 @@ def dimension_combinations_for(indicator)
 
     attr_reader :diabetes_enabled
 
-    memoize def daily_registrations_grouped_by_day
-      diagnosis = diabetes_enabled ? :all : :hypertension
-      RegisteredPatientsQuery.new.count_daily(facility, diagnosis: diagnosis, last: 30)
+    memoize def daily_total_follow_ups
+      records = Reports::FacilityDailyFollowUpAndRegistration.for_region(region).where("visit_date >= ?", DAYS_AGO.days.ago.to_date)
+      records.each_with_object({}) do |record, hsh|
+        hsh[record.period] = if region.diabetes_management_enabled?
+          record[:daily_follow_ups_htn_or_dm]
+        else
+          record[:daily_follow_ups_htn_only] + record[:daily_follow_ups_htn_and_dm]
+        end
+      end
+    end
+
+    memoize def daily_total_registrations
+      records = Reports::FacilityDailyFollowUpAndRegistration.for_region(region).where("visit_date >= ?", DAYS_AGO.days.ago.to_date)
+      records.each_with_object({}) do |record, hsh|
+        hsh[record.period] = if diabetes_enabled
+          record[:daily_registrations_htn_or_dm]
+        else
+          record[:daily_registrations_htn_only] + record[:daily_registrations_htn_and_dm]
+        end
+      end
+    end
+
+    def diagnosis_headers
+      {
+        hypertension: I18n.t("progress_tab.diagnoses.hypertension_only"),
+        diabetes: I18n.t("progress_tab.diagnoses.diabetes_only"),
+        hypertension_and_diabetes: I18n.t("progress_tab.diagnoses.hypertension_and_diabetes")
+      }
     end
```

Comment:
This is presentation logic that should go in `daily_progress_component_v2.html.erb`.

Reply:
yeah make sense